week3/feedback/feedback_client/main.go

```diff
@@ -0,0 +1,97 @@
+package main
+
+import (
+	"context"
+	"log"
+	"time"
+
+	"google.golang.org/grpc"
+
+	pb "grab/week3/feedback/feedback"
+)
+
+const (
+	address = "localhost:50051"
+)
+
+func addPassengerFeedback(ctx context.Context, c pb.CustomerFeedbackClient, in *pb.PassengerFeedback) {
+	resp, err := c.AddPassengerFeedback(ctx, &pb.AddPassengerFeedbackRequest{Feedback: in})
+	if err != nil {
+		log.Println("failed to add passenger feedback", err)
+		return
+	}
+	log.Println("success to add passenger feedback with msg:", resp.Msg)
+}
+
+func getFeedbacksByPassengerID(ctx context.Context, c pb.CustomerFeedbackClient, psgID int32) {
+	resp, err := c.GetFeedbacksByPassengerID(ctx, &pb.GetFeedbacksByPassengerIdRequest{PassengerID: psgID})
+	if err != nil {
+		log.Println("failed to get feedback by passengerID:", psgID, "with error:", err)
+		return
+	}
+	log.Println("feedback by passengerID", psgID, ":")
+	for _, feedback := range resp.Feedbacks {
+		log.Println(feedback)
+	}
+}
+
+func getFeedbackByBookingCode(ctx context.Context, c pb.CustomerFeedbackClient, bookingCode string) {
+	resp, err := c.GetFeedbackByBookingCode(ctx, &pb.GetFeedbackByBookingCodeRequest{BookingCode: bookingCode})
+	if err != nil {
+		log.Println("failed to get feedback by bookingCode", bookingCode, "with error:", err)
+		return
+	}
+	log.Println("feedback by bookingCode", bookingCode, ":")
+	log.Println(resp.Feedback)
+}
+
+func deleteFeedbacksByPassengerID(ctx context.Context, c pb.CustomerFeedbackClient, psgID int32) {
+	resp, err := c.DeleteFeedbacksByPassengerID(ctx, &pb.DeleteFeedbacksByPassengerIdRequest{PassengerID: psgID})
+	if err != nil {
+		log.Println("failed to delete feedbacks by passengerID", psgID, "with error:", err)
+		return
+	}
+	log.Println("deleted", resp.DeletedFeedbacks, "feedbacks of passengerID", psgID)
+}
+
+func simulateData(ctx context.Context, c pb.CustomerFeedbackClient) {
+	feedback1 := pb.PassengerFeedback{
```

Comment:
it's great if you can convert this to UTs (btw, 4th week will be about UT)

Reply:
Sure, looking forward to see how to write UT in go.